nvflare/fuel/f3/streaming/blob_streamer.py

```diff
@@ -49,18 +49,20 @@ def __init__(self, blob_cb: Callable):
         self.blob_cb = blob_cb
         self.size = 0
         self.buffer = None
+        self.pre_allocated = False
 
     def handle_blob_cb(self, future: StreamFuture, stream: Stream, resume: bool, *args, **kwargs) -> int:
 
         if resume:
             log.warning("Resume is not supported, ignored")
 
         self.size = stream.get_size()
+        self.pre_allocated = self.size > 0
 
-        if self.size > 0:
-            self.buffer = bytearray(self.size)
+        if self.pre_allocated:
+            self.buffer = wrap_view(bytearray(self.size))
         else:
-            self.buffer = bytes()
+            self.buffer = FastBuffer()
```

Comment:
what if we just do `self.buffer = bytearray()` ?

Reply:
That's what we had. It's getting slower and slower if we append to the bytearray. FastBuffer does over-allocation to reduce the number of allocations.